### PR TITLE
Stamina

### PR DESCRIPTION
**Pull request by a7c**

### CODE REVIEW

**Comment by a7c** [comment]
still need to do #33

File: src/App.tsx [hunk 1]
```diff
@@ -143,20 +147,50 @@ const Test = connect(
         onLearn: (item: model.Learnable) => dispatch(actions.learn(item)),
         onReview: (id: model.LearnableId, correct: boolean) =>
             dispatch(actions.review(id, correct)),
-        onWander: () => dispatch(actions.wander()),
+        onWander: () => {
```
Comment (lidavidm) [inline src/App.tsx]: Why not just create a `modifyResource` action dispatcher, then make `onMeditate` and `onWander` methods of `App`? That way you can pass `modifyResource` to other components that find it useful.

File: src/App.tsx [hunk 2]
```diff
@@ -143,20 +147,50 @@ const Test = connect(
...
-          <h1>Michael Mauer Simulator 2017</h1>
-          <Test/>
-      </div>
-    );
-  }
+interface AppProps {
+    dispatch: Dispatch<actions.Action>;
 }
 
+class BaseApp extends React.Component<AppProps> {
```
Comment (lidavidm) [inline src/App.tsx]: As I stated before, `index.tsx` actually has the store itself, and you can dispatch actions against that object. You don't need to do all this ceremony in React. Redux is its own library, and the store created in `index.tsx` has a dispatch method.

File: src/components/ActionPanel.tsx [hunk 3]
```diff
@@ -73,24 +85,35 @@ export default class ActionPanel extends React.Component<Props> {
...
-                         cost="-5 STA"
+                         benefit="+5 STA"
                          onClick={this.meditate}
                          paused={paused}
-                         cooldown={2000}
+                         cooldown={1000}
                      />
                      : false}
-                    <ActionButton label="Transliterate" benefit="+¥" paused={paused} />
+                    {learned.size ?
```
Comment (lidavidm) [inline src/components/ActionPanel.tsx]: I would rather this not appear until the quest is finished.

File: src/App.tsx (hunk 1 above)
Comment (a7c) [reply src/App.tsx]: addressed in 8410c77

File: src/App.tsx (hunk 2 above)
Comment (a7c) [reply src/App.tsx]: addressed in 56aa5f9

File: src/components/ActionPanel.tsx (hunk 3 above)
Comment (a7c) [reply src/components/ActionPanel.tsx]: @tyleretzel

Comment (a7c) [reply src/components/ActionPanel.tsx]: actually i changed it: 2a2c23b

File: src/components/ActionPanel.tsx [hunk 4]
```diff
@@ -64,33 +74,46 @@ export default class ActionPanel extends React.Component<Props> {
...
         return (
             <LabeledPanel title="Actions" id="actions">
                 <div>
                     <ActionButton
                         label="Wander"
-                        cost="-5 STA"
+                        cost={`${resources.WANDER_STA} STA`}
                         onClick={this.wander}
                         paused={paused}
+                        locked={stamina <= 0}
```
Comment (lidavidm) [inline src/components/ActionPanel.tsx]: This should be locked if STA is under `WANDER_STA`, not 0

Comment (a7c) [reply src/components/ActionPanel.tsx]: oops fixed lol  74ef89b

**Pull request merged**